### ISSUE TITLE
A 32GB machine ran out of memory importing data into Grakn

### ISSUE THREAD

**Issue by alexjpwalker** [vaticle/typedb#6173]
## Description

A 32GB machine ran out of memory importing data into Grakn.

## Environment

1. OS (where Grakn server runs): Linux in Azure
2. Grakn version (and platform): Grakn Core 2.0
3. Grakn client: client-python

## Reproducible Steps

Import the largest Biograkn dataset, which is a 1GB JSON file.

## Expected Output

Successful import

## Actual Output

After about an hour, Grakn suddenly terminated. The only log message was the single word "Killed." This suggests it was killed by the operating system.
 
## Additional information

The following log is the output of `dmesg -T| grep -E -i -B200 'killed process'`, which shows that the Linux's OOMKiller process killed Grakn:
```
[Mon Feb 15 18:05:04 2021] G1 Conc#0 invoked oom-killer: gfp_mask=0x100cca(GFP_HIGHUSER_MOVABLE), order=0, oom_score_adj=0
[Mon Feb 15 18:05:04 2021] CPU: 6 PID: 23664 Comm: G1 Conc#0 Not tainted 5.4.0-1039-azure #41~18.04.1-Ubuntu
[Mon Feb 15 18:05:04 2021] Hardware name: Microsoft Corporation Virtual Machine/Virtual Machine, BIOS 090008  12/07/2018
[Mon Feb 15 18:05:04 2021] Call Trace:
[Mon Feb 15 18:05:04 2021]  dump_stack+0x57/0x75
[Mon Feb 15 18:05:04 2021]  dump_header+0x4f/0x200
[Mon Feb 15 18:05:04 2021]  oom_kill_process+0xe6/0x120
[Mon Feb 15 18:05:04 2021]  out_of_memory+0x109/0x510
[Mon Feb 15 18:05:04 2021]  __alloc_pages_slowpath+0xa9b/0xe50
[Mon Feb 15 18:05:04 2021]  ? __switch_to_asm+0x34/0x70
[Mon Feb 15 18:05:04 2021]  __alloc_pages_nodemask+0x2cd/0x320
[Mon Feb 15 18:05:04 2021]  alloc_pages_current+0x6a/0xe0
[Mon Feb 15 18:05:04 2021]  __page_cache_alloc+0x6a/0xa0
[Mon Feb 15 18:05:04 2021]  pagecache_get_page+0xab/0x2c0
[Mon Feb 15 18:05:04 2021]  filemap_fault+0x669/0xb60
[Mon Feb 15 18:05:04 2021]  ? unlock_page_memcg+0x12/0x20
[Mon Feb 15 18:05:04 2021]  ? page_add_file_rmap+0x13a/0x180
[Mon Feb 15 18:05:04 2021]  ? filemap_map_pages+0x17d/0x3b0
[Mon Feb 15 18:05:04 2021]  ext4_filemap_fault+0x31/0x44
[Mon Feb 15 18:05:04 2021]  __do_fault+0x57/0x110
[Mon Feb 15 18:05:04 2021]  __handle_mm_fault+0xdf1/0x1330
[Mon Feb 15 18:05:04 2021]  ? __switch_to_asm+0x40/0x70
[Mon Feb 15 18:05:04 2021]  ? __switch_to_asm+0x40/0x70
[Mon Feb 15 18:05:04 2021]  handle_mm_fault+0xb7/0x200
[Mon Feb 15 18:05:04 2021]  __do_page_fault+0x29c/0x4c0
[Mon Feb 15 18:05:04 2021]  do_page_fault+0x35/0x110
[Mon Feb 15 18:05:04 2021]  page_fault+0x39/0x40
[Mon Feb 15 18:05:04 2021] RIP: 0033:0x7f619b1a0f20
[Mon Feb 15 18:05:04 2021] Code: Bad RIP value.
[Mon Feb 15 18:05:04 2021] RSP: 002b:00007f6198137c38 EFLAGS: 00010206
[Mon Feb 15 18:05:04 2021] RAX: 0000000000000004 RBX: 000000000000618d RCX: 00007f619bef3660
[Mon Feb 15 18:05:04 2021] RDX: 00000008003b85f0 RSI: 00000006888f0c78 RDI: 00007f6198137db8
[Mon Feb 15 18:05:04 2021] RBP: 00007f6198137d40 R08: 000000060a200000 R09: 000000000fcde18f
[Mon Feb 15 18:05:04 2021] R10: 00007f6165603000 R11: 00000000003f3786 R12: 00007f619bef422c
[Mon Feb 15 18:05:04 2021] R13: 00000006888f0c78 R14: 000000000000000e R15: 00007f6198137da0
[Mon Feb 15 18:05:04 2021] Mem-Info:
[Mon Feb 15 18:05:04 2021] active_anon:8083198 inactive_anon:84 isolated_anon:0
                            active_file:140 inactive_file:0 isolated_file:0
                            unevictable:0 dirty:2 writeback:0 unstable:0
                            slab_reclaimable:12537 slab_unreclaimable:24436
                            mapped:286 shmem:181 pagetables:17644 bounce:0
                            free:51947 free_pcp:2107 free_cma:0
[Mon Feb 15 18:05:04 2021] Node 0 active_anon:32332792kB inactive_anon:336kB active_file:560kB inactive_file:0kB unevictable:0kB isolated(anon):0kB isolated(file):0kB mapped:1144kB dirty:8kB writeback:0kB shmem:724kB shmem_thp: 0kB shmem_pmdmapped: 0kB anon_thp: 9684992kB writeback_tmp:0kB unstable:0kB all_unreclaimable? no
[Mon Feb 15 18:05:04 2021] Node 0 DMA free:15908kB min:32kB low:44kB high:56kB active_anon:0kB inactive_anon:0kB active_file:0kB inactive_file:0kB unevictable:0kB writepending:0kB present:15992kB managed:15908kB mlocked:0kB kernel_stack:0kB pagetables:0kB bounce:0kB free_pcp:0kB local_pcp:0kB free_cma:0kB
[Mon Feb 15 18:05:04 2021] lowmem_reserve[]: 0 915 32065 32065 32065
[Mon Feb 15 18:05:04 2021] Node 0 DMA32 free:126444kB min:1928kB low:2864kB high:3800kB active_anon:837740kB inactive_anon:0kB active_file:4kB inactive_file:20kB unevictable:0kB writepending:0kB present:1032128kB managed:966592kB mlocked:0kB kernel_stack:0kB pagetables:1632kB bounce:0kB free_pcp:608kB local_pcp:4kB free_cma:0kB
[Mon Feb 15 18:05:04 2021] lowmem_reserve[]: 0 0 31149 31149 31149
[Mon Feb 15 18:05:04 2021] Node 0 Normal free:65436kB min:65620kB low:97516kB high:129412kB active_anon:31495052kB inactive_anon:336kB active_file:328kB inactive_file:0kB unevictable:0kB writepending:8kB present:32505856kB managed:31905444kB mlocked:0kB kernel_stack:5712kB pagetables:68944kB bounce:0kB free_pcp:7820kB local_pcp:228kB free_cma:0kB
[Mon Feb 15 18:05:04 2021] lowmem_reserve[]: 0 0 0 0 0
[Mon Feb 15 18:05:04 2021] Node 0 DMA: 1*4kB (U) 0*8kB 0*16kB 1*32kB (U) 2*64kB (U) 1*128kB (U) 1*256kB (U) 0*512kB 1*1024kB (U) 1*2048kB (M) 3*4096kB (M) = 15908kB
[Mon Feb 15 18:05:04 2021] Node 0 DMA32: 74*4kB (UM) 49*8kB (UM) 19*16kB (U) 14*32kB (UM) 9*64kB (UM) 3*128kB (U) 1*256kB (U) 0*512kB 1*1024kB (M) 0*2048kB 30*4096kB (M) = 126560kB
[Mon Feb 15 18:05:04 2021] Node 0 Normal: 1877*4kB (UME) 1107*8kB (UME) 811*16kB (UME) 428*32kB (UME) 204*64kB (UME) 67*128kB (UME) 9*256kB (ME) 0*512kB 0*1024kB 0*2048kB 0*4096kB = 66972kB
[Mon Feb 15 18:05:04 2021] Node 0 hugepages_total=0 hugepages_free=0 hugepages_surp=0 hugepages_size=1048576kB
[Mon Feb 15 18:05:04 2021] Node 0 hugepages_total=0 hugepages_free=0 hugepages_surp=0 hugepages_size=2048kB
[Mon Feb 15 18:05:04 2021] 613 total pagecache pages
[Mon Feb 15 18:05:04 2021] 0 pages in swap cache
[Mon Feb 15 18:05:04 2021] Swap cache stats: add 0, delete 0, find 0/0
[Mon Feb 15 18:05:04 2021] Free swap  = 0kB
[Mon Feb 15 18:05:04 2021] Total swap = 0kB
[Mon Feb 15 18:05:04 2021] 8388494 pages RAM
[Mon Feb 15 18:05:04 2021] 0 pages HighMem/MovableOnly
[Mon Feb 15 18:05:04 2021] 166508 pages reserved
[Mon Feb 15 18:05:04 2021] 0 pages cma reserved
[Mon Feb 15 18:05:04 2021] 0 pages hwpoisoned
[Mon Feb 15 18:05:04 2021] Tasks state (memory values in pages):
[Mon Feb 15 18:05:04 2021] [  pid  ]   uid  tgid total_vm      rss pgtables_bytes swapents oom_score_adj name
[Mon Feb 15 18:05:04 2021] [    506]     0   506    25919      336   212992        0             0 systemd-journal
[Mon Feb 15 18:05:04 2021] [    510]     0   510    24428       45   102400        0             0 lvmetad
[Mon Feb 15 18:05:04 2021] [    521]     0   521    10927      277   110592        0         -1000 systemd-udevd
[Mon Feb 15 18:05:04 2021] [    578]     0   578     3161      366    69632        0             0 hv_kvp_daemon
[Mon Feb 15 18:05:04 2021] [    770] 62583   770    35490      149   184320        0             0 systemd-timesyn
[Mon Feb 15 18:05:04 2021] [    854]   100   854    20055      194   167936        0             0 systemd-network
[Mon Feb 15 18:05:04 2021] [    882]   101   882    17693      175   180224        0             0 systemd-resolve
[Mon Feb 15 18:05:04 2021] [   1092]     0  1092    77204       95    98304        0             0 lxcfs
[Mon Feb 15 18:05:04 2021] [   1096]     0  1096    27623       86   126976        0             0 irqbalance
[Mon Feb 15 18:05:04 2021] [   1104]     0  1104    42707     1947   225280        0             0 networkd-dispat
[Mon Feb 15 18:05:04 2021] [   1109]     0  1109    19950     3119   204800        0             0 python3
[Mon Feb 15 18:05:04 2021] [   1127]     0  1127    17669      195   172032        0             0 systemd-logind
[Mon Feb 15 18:05:04 2021] [   1146]   102  1146    66818     1733   180224        0             0 rsyslogd
[Mon Feb 15 18:05:04 2021] [   1159]     0  1159     7084       52   102400        0             0 atd
[Mon Feb 15 18:05:04 2021] [   1195]   103  1195    12513      175   139264        0          -900 dbus-daemon
[Mon Feb 15 18:05:04 2021] [   1248]     0  1248    72028      257   196608        0             0 accounts-daemon
[Mon Feb 15 18:05:04 2021] [   1256]     0  1256     7938       73   106496        0             0 cron
[Mon Feb 15 18:05:04 2021] [   1284]     0  1284    18076      188   184320        0         -1000 sshd
[Mon Feb 15 18:05:04 2021] [   1307]     0  1307    46919     1977   262144        0             0 unattended-upgr
[Mon Feb 15 18:05:04 2021] [   1330]     0  1330    72221      706   217088        0             0 polkitd
[Mon Feb 15 18:05:04 2021] [   1344]     0  1344     4104       36    73728        0             0 agetty
[Mon Feb 15 18:05:04 2021] [   1362]     0  1362     3723       33    69632        0             0 agetty
[Mon Feb 15 18:05:04 2021] [   1384]     0  1384    96983     5071   270336        0             0 python3
[Mon Feb 15 18:05:04 2021] [   1572]     0  1572     1128       17    57344        0             0 none
[Mon Feb 15 18:05:04 2021] [  23448]  1000 23448    19176      262   196608        0             0 systemd
[Mon Feb 15 18:05:04 2021] [  23449]  1000 23449    27973      630   245760        0             0 (sd-pam)
[Mon Feb 15 18:05:04 2021] [  23543]     0 23543    26997      255   262144        0             0 sshd
[Mon Feb 15 18:05:04 2021] [  23625]  1000 23625    26997      253   253952        0             0 sshd
[Mon Feb 15 18:05:04 2021] [  23626]  1000 23626     5804      447    86016        0             0 bash
[Mon Feb 15 18:05:04 2021] [  23655]  1000 23655  7926346  6673428 54706176        0             0 java
[Mon Feb 15 18:05:04 2021] [  23742]     0 23742    26997      254   249856        0             0 sshd
[Mon Feb 15 18:05:04 2021] [  23826]  1000 23826    27129      360   245760        0             0 sshd
[Mon Feb 15 18:05:04 2021] [  23827]  1000 23827     5804      425    81920        0             0 bash
[Mon Feb 15 18:05:04 2021] [  23918]  1000 23918  2395225  1342345 11632640        0             0 python3
[Mon Feb 15 18:05:04 2021] [   2667]     0  2667    26997      255   258048        0             0 sshd
[Mon Feb 15 18:05:04 2021] [   2795]  1000  2795    26997      253   245760        0             0 sshd
[Mon Feb 15 18:05:04 2021] [   2798]  1000  2798     5804      447    90112        0             0 bash
[Mon Feb 15 18:05:04 2021] [   3217]  1000  3217  3082472    44917   929792        0             0 java
[Mon Feb 15 18:05:04 2021] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice,task=java,pid=23655,uid=1000
[Mon Feb 15 18:05:04 2021] Out of memory: Killed process 23655 (java) total-vm:31705384kB, anon-rss:26693712kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:53424kB oom_score_adj:0
```
(Grakn also died in a similar manner when Tomas tried running it on his own machine)

**Comment by alexjpwalker** [comment]
Just noticed this was erroneously in backlog

**Comment by chris-aeviator** [comment]
I'm running into this with a 64 GB Ram machine importing (via node.js client) 75k entries

**Comment by flyingsilverfin** [comment]
We expect this to be resolved by now (TypeDB 2.6), please reopen if it still an issue!
